addons/sale_payment/models/payment.py

```diff
@@ -63,7 +63,9 @@ def _confirm_so(self):
             _logger.info('<%s> transaction authorized, auto-confirming order %s (ID %s)', self.acquirer_id.provider, self.sale_order_id.name, self.sale_order_id.id)
             if self.sale_order_id.state in ('draft', 'sent'):
                 self.sale_order_id.with_context(send_email=True).action_confirm()
-        elif self.state == 'done':
+            return 'authorized_sale'
+
+        if self.state == 'done':
```

Comment:
if-elif ?

Reply:
No, even if the condition of the first "if" is true, the next "if" must be checked too